### PR TITLE
fixed cell initialization in LSTM layer

### PR DESCRIPTION
**Pull request by beniz**
This fix allows LSTM to work correctly. 

The layer now works properly, and could be activated in https://github.com/Tencent/ncnn/blob/master/src/CMakeLists.txt#L162

This is a fix by @fantes.

See #697

### CODE REVIEW

File: tools/caffe/caffe.proto [hunk 1]
```diff
@@ -359,15 +525,17 @@ message LayerParameter {
   // engine parameter for selecting the implementation.
   // The default for the engine is set by the ENGINE switch at compile-time.
   optional AccuracyParameter accuracy_param = 102;
+  optional AnnotatedDataParameter annotated_data_param = 200;
   optional ArgMaxParameter argmax_param = 103;
   optional BatchNormParameter batch_norm_param = 139;
   optional BiasParameter bias_param = 141;
-  optional BNParameter bn_param = 45;
```
Comment (nihui) [inline tools/caffe/caffe.proto]: do not drop BN layer for backward compability

File: tools/caffe/caffe.proto [hunk 2]
```diff
@@ -594,17 +970,6 @@ message BiasParameter {
   optional FillerParameter filler = 3;
 }
 
-// Message that stores parameters used by BN (Batch Normalization)  layer
-message BNParameter {
```
Comment (nihui) [inline tools/caffe/caffe.proto]: do not drop BN layer for backward compability

File: tools/caffe/caffe2ncnn.cpp [hunk 3]
```diff
@@ -321,18 +321,23 @@ static bool read_proto_from_binary(const char* filepath, google::protobuf::Messa
 
 int main(int argc, char** argv)
 {
-    if (!(argc == 3 || argc == 5 || argc == 6 || argc == 7))
+    int ocr = 0;
+
+    if (!(argc == 4 || argc == 6 || argc == 7 || argc == 8))
     {
-        fprintf(stderr, "Usage: %s [caffeproto] [caffemodel] [ncnnproto] [ncnnbin] [quantizelevel] [int8scaletable]\n", argv[0]);
+        fprintf(stderr, "Usage: %s ocr|noocr [caffeproto] [caffemodel] [ncnnproto] [ncnnbin] [quantizelevel] [int8scaletable]\n", argv[0]);
```
Comment (nihui) [inline tools/caffe/caffe2ncnn.cpp]: new optional parameter should always be placed as the last one for keeping the old usage works

File: src/layer/continuationindicator.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+//
+// https://opensource.org/licenses/BSD-3-Clause
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include "continuationindicator.h"
+#include <iostream>
```
Comment (nihui) [inline src/layer/continuationindicator.cpp]: do not include the iostream header, no usecase

**Comment by beniz** [comment]
@nihui apologies, we modified our local branch afterwards, thus affecting this PR. The fix is actually a one liner. Let's close it then we'll resubmit.

**Comment by beniz** [comment]
See #784

**Pull request closed without merging**